domains/transfer-agreements/TransferAgreements.API/API/Models/Connection.cs

```diff
@@ -0,0 +1,15 @@
+using System;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Models;
+
+[Index(nameof(CompanyAId))]
+[Index(nameof(CompanyBId))]
```

Comment:
Here annotations are used, but in ApplicationDbContext, the fluent API is used for something similar. I suggest that you pick only one approach.